src/NHibernate.Test/NHSpecificTest/DtcFailures/DtcFailuresFixture.cs

```diff
@@ -266,69 +327,107 @@ public void CanDeleteItemInDtc()
 			object id;
 			using (var tx = new TransactionScope())
 			{
-				using (ISession s = sessions.OpenSession())
+				using (var s = sessions.OpenSession())
 				{
-					id = s.Save(new Person {CreatedAt = DateTime.Today});
+					id = s.Save(new Person { CreatedAt = DateTime.Today });
 
-					new ForceEscalationToDistributedTx();
+					ForceEscalationToDistributedTx.Escalate();
 
 					tx.Complete();
 				}
 			}
 
+			using (var s = sessions.OpenSession())
+			using (s.BeginTransaction())
+			{
+				Assert.AreEqual(1, s.Query<Person>().Count(), "Entity not found in database.");
+			}
+
 			using (var tx = new TransactionScope())
 			{
-				using (ISession s = sessions.OpenSession())
+				using (var s = sessions.OpenSession())
 				{
-					new ForceEscalationToDistributedTx();
+					ForceEscalationToDistributedTx.Escalate();
 
 					s.Delete(s.Get<Person>(id));
 
 					tx.Complete();
 				}
 			}
+
+			AssertNoPersons();
 		}
 
 		[Test]
 		[Description("Open/Close a session inside a TransactionScope fails.")]
 		public void NH1744()
 		{
-			using (var tx = new TransactionScope())
+			using (new TransactionScope())
 			{
-				using (ISession s = sessions.OpenSession())
+				using (var s = sessions.OpenSession())
 				{
 					s.Flush();
 				}
 
-				using (ISession s = sessions.OpenSession())
+				using (var s = sessions.OpenSession())
 				{
 					s.Flush();
 				}
 
-				//and I always leave the transaction disposed without calling tx.Complete(), I let the database server to rollback all actions in this test. 
-			} 
+				//and I always leave the transaction disposed without calling tx.Complete(), I let the database server to rollback all actions in this test.
+			}
+		}
+
+		[Test]
+		public void CanUseSessionOutsideOfScopeAfterScope()
+		{
+			using (var s = sessions.WithOptions().ConnectionReleaseMode(ConnectionReleaseMode.OnClose).OpenSession())
+			{
+				using (var tx = new TransactionScope())
+				{
+					s.Save(new Person { CreatedAt = DateTime.Today });
+
+					ForceEscalationToDistributedTx.Escalate();
+
+					tx.Complete();
+				}
+
+				Assert.AreEqual(1, s.Query<Person>().Count(), "Entity not found in database.");
```

Comment:
The custom transaction provider was enlisting with null for ensuring next session usages will not try going on with the completed transaction. This branch does not do that yet. This test is here for checking if it is required.